src/main/java/cms/gov/madie/measure/services/VersionService.java

```diff
@@ -114,6 +116,15 @@ public Measure createDraft(String id, String measureName, String username) {
     return savedDraft;
   }
 
+  private List<Group> updateGroupIds(List<Group> groups) {
+    if (groups != null) {
+      List<Group> updatedGroups = groups;
+      updatedGroups.forEach(group -> group.setId(ObjectId.get().toString()));
+      return updatedGroups;
```

Comment:
You are creating a new reference to an existing variable. you are still modifying existing objects. try this

```
return groups.stream()
          .map(group -> group.toBuilder().id(ObjectId.get().toString()).build())
          .collect(Collectors.toList());
```
Also, I think the method name is not so appropriate to the task it is doing. Probably `cloneMeasureGroups(List<Group> groups)  ` or something similar

Reply:
Thank you, I have updated the code